animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -370,3 +384,272 @@ def calculateLinkedVertices(int verticesAmount, EdgeIndicesList edges):
+
+    cdef Vector3 preVertex, earVertex, nexVertex, ab, bc, v1, v2, v3
+    cdef LongList neighbors, indices, angles, sortAngles
+    cdef Vector3DList polyVertices
+    cdef float angle
+    cdef Py_ssize_t polyStart, triIndex, polyIndex
+    cdef Py_ssize_t j, k, amount, index, preIndex, earIndex, nexIndex
+
+    triIndex = 0
+    polyIndex = 0
+    for i in range(polygonAmount):
+        polyLength = _oldPolyLengths[i]
+        polyStart = _oldPolyStarts[i]
+        neighbors = LongList(length = polyLength)
+        indices = LongList(length = polyLength)
+        for j in range(polyLength):
+            neighbors.data[j] = _oldIndices[polyStart + j]
+            indices.data[j] = j
+
+        polyVertices = transformVerticesOfPolygon(oldVertices, neighbors)
+        if not polyCounterClockwise(polyVertices):
+            neighbors = neighbors.reversed()
+            polyVertices = polyVertices.reversed()
+
+        for j in range(polyLength):
+            amount = indices.length
+
+            if amount == 3:
+                _newPolyStarts[triIndex] = polyIndex
+                _newPolyLengths[triIndex] = 3
+                triIndex += 1
+
+                _newIndices[polyIndex] = neighbors.data[indices.data[0]]
+                _newIndices[polyIndex + 1] = neighbors.data[indices.data[1]]
+                _newIndices[polyIndex + 2] = neighbors.data[indices.data[2]]
+                polyIndex += 3
+
+                break
+
+            # Calculate inner-angle for all vertices of polygon
+            angles = LongList(length = amount)
+            for k in range(amount):
+                preVertex = polyVertices.data[indices.data[(amount + k - 1) % amount]]
+                earVertex = polyVertices.data[indices.data[k]]
+                nexVertex = polyVertices.data[indices.data[(k + 1) % amount]]
+
+                subVec3(&ab, &earVertex, &preVertex)
+                subVec3(&bc, &earVertex, &nexVertex)
+
+                normalizeVec3_InPlace(&ab)
+                normalizeVec3_InPlace(&bc)
+
+                angle = angleNormalizedVec3(&ab, &bc)
+                angles.data[k] = int((angle - angle % 0.001) * 180.0 / pi)
+            sortAngles = LongList.fromNumpyArray(numpy.sort(angles.asMemoryView()).astype(int))
+
+            # Removing an ear which has smallest inner-angle
+            for k in range(amount):
+                index = angles.index(sortAngles[k])
+                preIndex = indices.data[(amount + index - 1) % amount]
+                earIndex = indices.data[index]
+                nexIndex = indices.data[(index + 1) % amount]
+
+                v1 = polyVertices.data[preIndex]
+                v2 = polyVertices.data[earIndex]
+                v3 = polyVertices.data[nexIndex]
+                if convexVertex(v1, v2, v3) and not pointsInTriangle(polyVertices, v1, v2, v3, preIndex, earIndex, nexIndex):
+                    indices = LongList.fromNumpyArray(numpy.delete(indices.asMemoryView(), index).astype(int))
```

Comment:
Same as I mentioned above.

Reply:
Is there a way to delete an element Or should I use loop for that?

Reply:
The Animation Nodes list structures implement element deletion, so you can just do that. Either using the `del` keyword or the `removeValueAtIndex` method.
But if you can avoid deletions, that would be best. Perhaps by keeping track of what was removed without actually removing it.

Reply:
I have modified the algorithm, now no need for deletion.